source/common/network/address_impl.cc

```diff
@@ -130,7 +139,15 @@ int Ipv4Instance::connect(int fd) const {
 }
 
 int Ipv4Instance::socket(SocketType type) const {
-  return ::socket(AF_INET, flagsFromSocketType(type), 0);
+  int fd = ::socket(AF_INET, flagsFromSocketType(type), 0);
+  RELEASE_ASSERT(fd != -1);
+
+#ifdef __APPLE__
+  // Cannot set SOCK_NONBLOCK as a ::socket flag
+  RELEASE_ASSERT(fcntl(fd, F_SETFL, O_NONBLOCK) != -1);
```

Comment:
Hmm... I'm confused. Apparently, it doesn't overwrite flags on OS X.

`fcntl(fd, F_GETFL)` on a new `socket` returns `0x0002` (`O_RDWR`), but it returns `0x0006` (`O_RDWR|O_NONBLOCK`) after `fcntl(fd, F_SETFL, O_NONBLOCK)`, which is not something I'd expect.